src/textual/app.py

```diff
@@ -529,6 +534,25 @@ def return_value(self) -> ReturnType | None:
         """
         return self._return_value
 
+    @property
+    def return_code(self) -> int | None:
+        """The return code with which the app exited.
+
+        Non-zero codes indicate errors.
+        A value of 1 means the app exited with a fatal error.
+        If the app hasn't run yet or if it is running, this will be `None`.
+
+        Example:
+            The return code can be used to exit the process via `sys.exit`.
+            ```py
+            my_app.run()
+            sys.exit(my_app.return_code)
+            ```
+        """
+        if not self._started_running or self.is_running:
```

Comment:
So `app.return_code` will be None, immediately after calling `exit(return_code=10)` ?

I think this is simpler than you imagine. Suggest you set the return code to `1` in `App._handle_exception`. All errors will go through there.

Reply:
Done in ecd7c93.